UnwrapBlog.md

```diff
@@ -0,0 +1,92 @@
+Advanced TypeScript type FTW!
+
+In Pulumi we have a data model that allows people to express complex data that may *eventually* be available. Traditional JavaScript programming might expose that as a Promise<T>, but we've taken that one step further by introducing a type we call:
+
+```ts
+type Input<T> = T | Promise<T> | Output<T>;
+
+// Like a Promise<T>, but also keeps track of Dependency information so we can
+// figure out what the upstream/downstream impact is of any value change.
+interface Output<T> { /* ... */ }
+```
+
+This complex type can often show up in forms like:
+
+```ts
+interface VpcInput {
+    ipAddresses: Input<Input<string>[]>;
+}
+```
+
+Breaking that down, it's basically saying "I'll take an array of strings.  But those actual string values can individually be strings, Promises-of-strings, or Strings-carrying-dependency information." It even goes one step beyond that, because the entire array itself could be just an array-of-those-things, or a Promise-of-array-of-those-things, or an Array-of-those-things-with-dependency information.  Whew...!
+
+This approach is great for people producing the data.  As a producer I could do:
+
+```ts
+new Vpc({ ipAddress: ["an-address"] }); // or even 
```

Comment:
How is the first of there related to the type above?  This seems like it wouldn't work.

Reply:
An array of strings, is itself an ```Input<Input<string>[]>```

Reply:
oh, i see.  this is about saying ipAddress.  yes, that was a typo.  fixed!